clippy_lints/src/transmute.rs

```diff
@@ -122,7 +122,12 @@ impl LateLintPass for Transmute {
 
 
                                     let sugg = if from_pty.ty == to_rty.ty {
-                                        format!("{}{}", deref, arg)
+                                        // Put things in parentheses if they are more complex
+                                        if let ExprPath(..) = args[0].node {
```

Comment:
Would be nice if we had a precendence tool that could figure this out on its own 😄